additionalResolvers/flashcard.ts

```diff
@@ -0,0 +1,277 @@
+                                          skillPoints
+                                          skillTypes
+                                    }
+                                    items{
+                                        id
+                                        associatedSkills{id,skillName}
+                                        associatedBloomLevels
+                                    }
+                             }
+                            
+                    }
+                    `,
+                    context,
+                    info
+                });
+            let assessment =assessments[0];
+            let oldItems= assessment.items;
+            let items=assessment.items;
+            items=[...items,_args.item];
+            assessment.items=items;
+            let selectionSet=`{updateAssessment(input: {
+                metadata: {
+                    name: "${assessment.metadata.name}",
+                    suggestedDate: "${assessment.metadata.suggestedDate}",
+                    chapterId: "${assessment.metadata.chapterId}",
+                    rewardPoints: ${assessment.metadata.rewardPoints},
+                    tagNames: [${assessment.metadata.tagNames.map(tag => `"${tag}"`)}]
+                },
+                assessmentMetadata: {
+                    skillPoints: ${assessment.assessmentMetadata.skillPoints},
+                    skillTypes: [${assessment.assessmentMetadata.skillTypes.map(skillType => `${skillType}`)}],
+                    initialLearningInterval: ${assessment.assessmentMetadata.initialLearningInterval}
+                },
+                items:[
+                    ${assessment.items.map(item => `{
+                        ${item.id != null ? `id:"${item.id}",` : ''}
+                        associatedSkills:[${item.associatedSkills.map(skill => `{ ${skill.id != null ? `id:"${skill.id}",` : ''} skillName:"${skill.skillName}"}`)}],
+                        associatedBloomLevels:[${item.associatedBloomLevels.map(level => `${level}`)}]
+                    }`)}
+                ]
+            }) {
+                id
+                items{
+                    id
+                    associatedSkills{
+                        id
+                        skillName
+                    }
+                    associatedBloomLevels
+                }
+            }
+            }`;
+            let updatedItems = await context.ContentService.Mutation.mutateContent({
+                root,
+                args:{contentId:_args.assessmentId},
+                selectionSet:selectionSet, 
+                context,
+                info});
+
+            let flashcardInput=_args.flashcardInput;
+            for (let item of updatedItems.updateAssessment.items) {
+                if (!oldItems.some(oldItem => oldItem.id === item.id)) {
+                    flashcardInput.itemId= item.id;
+                    break;
+                }
+            }
```

Comment:
Nitpick: This could be written in the following way, which would be a bit more concise & intuitive to understand what's happening, at least to me:

```js
flashcardInput.itemId = updatedItems.updateAssessment.items
.find(item => !oldItems.some(oldItem.id === item.id))
.id;
```